src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java

```diff
@@ -86,19 +88,85 @@
   public static final Type TYPE_UNSIGNED_LONG = TypeParser.createFrom("unsigned long", true);
 
   private static final Logger LOGGER = LoggerFactory.getLogger(CXXLanguageFrontend.class);
-  private static final IncludeFileContentProvider INCLUDE_FILE_PROVIDER =
+  private final IncludeFileContentProvider includeFileContentProvider =
       new InternalFileContentProvider() {
         @Nullable
         private InternalFileContent getContentUncached(String path) {
           if (!getInclusionExists(path)) {
             LOGGER.debug("Include file not found: {}", path);
             return null;
           }
+
+          // check, if the file is on the blacklist
+          if (absoluteOrRelativePathIsInList(path, config.includeBlacklist)) {
+            LOGGER.debug("Blacklisting include file: {}", path);
+            return null;
+          }
+
+          // check, if the white-list exists at all
+          if (hasWhitelist()
+              &&
+              // and ignore the file if it is not on the whitelist
+              !absoluteOrRelativePathIsInList(path, config.includeWhitelist)) {
+            LOGGER.debug("Include file {} not on the whitelist. Ignoring.", path);
+            return null;
+          }
+
           LOGGER.debug("Loading include file {}", path);
           FileContent content = FileContent.createForExternalFileLocation(path);
           return (InternalFileContent) content;
         }
 
+        private boolean hasWhitelist() {
```

Comment:
Propose to name it `hasIncludeWhitelist` as otherwise it will be unclear what "whitelist" refers to.